transformer_engine/pytorch/csrc/extensions.cu

```diff
@@ -447,14 +441,10 @@ at::Tensor cast_to_fp8(const at::Tensor &input,
 
     auto output = at::empty_like(input, at::CUDA(GetATenDType(otype)));
 
-    auto input_cu     = makeTransformerEngineTensor(input);
+    auto input_cu     = makeTransformerEngineTensor(input, amax, scale, scale_inv);
     auto output_cu    = makeTransformerEngineTensor(output.data_ptr(), {N, H}, otype);
```

Comment:
```suggestion
    auto input_cu     = makeTransformerEngineTensor(input);
    auto output_cu    = makeTransformerEngineTensor(output.data_ptr(), {N, H}, otype,
                                                    amax.data_ptr(), scale.data_ptr(),
                                                    scale_inv.data_ptr());
```

Reply:
`cast_to_fp8` has high precision input and FP8 output. The amax, scale and inverse of scale data correspond on the output side, not input. The above suggestion fixes the issue.